flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/TemporalRowTimeJoinOperator.java

```diff
@@ -209,6 +209,7 @@ public void onEventTime(InternalTimer<Object, VoidNamespace> timer) throws Excep
                 registerProcessingCleanupTimer();
             } else {
                 cleanupLastTimer();
+                nextLeftIndex.clear();
```

Comment:
why change here?

Reply:
When the program is going into line#211, that means leftState is empty and rightState is empty, so cleanupLastTimer() is to delete the cleanup trigger for this key. At this time, nextLeftIndexState is not empty(registeredTimer is empty because line#201), the nextLeftIndexState will be stored forever because clean update trigger have been deleted.

Reply:
Look forward to your soonest reply. @leonardBang